src/System.Private.ServiceModel/src/System/ServiceModel/Channels/CoreClrSocketConnection.cs

```diff
@@ -580,6 +594,20 @@ protected override int ReadCore(byte[] buffer, int offset, int size, TimeSpan ti
             return bytesRead;
         }
 
+        protected override void TraceSocketReadStop(int bytesRead, bool async)
+        {
+            var remoteEndpoint = (IPEndPoint)_socket.RemoteEndPoint;
+            string remoteEndpointAddressString = remoteEndpoint.Address + ":" + remoteEndpoint.Port;
+            if (!async)
+            {
+                WcfEventSource.Instance.SocketReadStop((_socket != null) ? _socket.GetHashCode() : -1, bytesRead, remoteEndpointAddressString);
+            }
```

Comment:
Hmmm, it checks _socket for null here but not when it reads _socket.RemoteEndpoint above?  When can it be null?